pkgs/by-name/wp/wpcleaner/package.nix

```diff
@@ -0,0 +1,116 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeDesktopItem,
+  copyDesktopItems,
+  makeWrapper,
+  jdk17,
+  ant,
+  stripJavaArchivesHook,
+  gettext,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wpcleaner";
+  version = "2.0.5-unstable-2025-04-25";
+  src = fetchFromGitHub {
+    owner = "WPCleaner";
+    repo = "wpcleaner";
+    rev = "7fd357cf26349658183517658139870dd45eaedc";
+    hash = "sha256-iaAP/5Z+ghvMAn4ke7lhRqKov/3jXr0LMwbPDZ052j0=";
+  };
+
+  wpcleanerJar = "$out/share/wpcleaner/WikipediaCleaner.jar";
+  clientScript = "$out/bin/wpcleaner";
+  botScript = "$out/bin/wpcleaner-bot";
+  runTaskScript = "$out/bin/wpcleaner-run-task";
```

Comment:
Perhaps it would be a good idea to move out those attributes in a `let/in` statement? So we can clearly identify what are the attributes belonging to `stdenv.mkDerivation` and the rest.

Reply:
I thought that `let/in` statements should be avoided, as they cannot be overridden (cf. [https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293/4 for instance](https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293/4).

Reply:
I would still use a let/in here for all these "variables" that you need, other attributes will remain "overridable" if necessary, no worries.